exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1494 @@
+		if err != nil {
+			return order.Detail{}, err
+		}
+
+		if len(resp) != 1 {
+			return order.Detail{}, fmt.Errorf("invalid order's count found")
+		}
+
+		return order.Detail{
+			Amount:         resp[0].Qty,
+			Exchange:       by.Name,
+			ID:             resp[0].OrderID,
+			ClientOrderID:  resp[0].OrderLinkID,
+			Side:           order.Side(resp[0].Side),
+			Type:           order.Type(resp[0].OrderType),
+			Pair:           pair,
+			Cost:           resp[0].CumulativeQty,
+			AssetType:      assetType,
+			Status:         order.Status(resp[0].OrderStatus),
+			Price:          resp[0].Price,
+			ExecutedAmount: resp[0].Qty - resp[0].LeavesQty,
+			Date:           resp[0].CreatedAt,
+			LastUpdated:    resp[0].UpdatedAt,
+		}, nil
+
+	case asset.Futures:
+		resp, err := by.GetActiveRealtimeOrders(ctx, pair, orderID, "")
+		if err != nil {
+			return order.Detail{}, err
+		}
+
+		if len(resp) != 1 {
+			return order.Detail{}, fmt.Errorf("invalid order's count found")
+		}
+
+		return order.Detail{
+			Amount:         resp[0].Qty,
+			Exchange:       by.Name,
+			ID:             resp[0].OrderID,
+			ClientOrderID:  resp[0].OrderLinkID,
+			Side:           order.Side(resp[0].Side),
+			Type:           order.Type(resp[0].OrderType),
+			Pair:           pair,
+			Cost:           resp[0].CumulativeQty,
+			AssetType:      assetType,
+			Status:         order.Status(resp[0].OrderStatus),
+			Price:          resp[0].Price,
+			ExecutedAmount: resp[0].Qty - resp[0].LeavesQty,
+			Date:           resp[0].CreatedAt,
+			LastUpdated:    resp[0].UpdatedAt,
+		}, nil
+
+	default:
+		return order.Detail{}, fmt.Errorf("assetType not supported: %v", assetType)
+	}
+}
+
+// GetDepositAddress returns a deposit address for a specified currency
+func (by *Bybit) GetDepositAddress(ctx context.Context, cryptocurrency currency.Code, _, chain string) (*deposit.Address, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// WithdrawCryptocurrencyFunds returns a withdrawal ID when a withdrawal is
+// submitted
+func (by *Bybit) WithdrawCryptocurrencyFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// WithdrawFiatFunds returns a withdrawal ID when a withdrawal is
+// submitted
+func (by *Bybit) WithdrawFiatFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// WithdrawFiatFundsToInternationalBank returns a withdrawal ID when a withdrawal is
+// submitted
+func (by *Bybit) WithdrawFiatFundsToInternationalBank(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetActiveOrders retrieves any orders that are active/open
+func (by *Bybit) GetActiveOrders(ctx context.Context, req *order.GetOrdersRequest) ([]order.Detail, error) {
```

Comment:
There are problems here:

```json
[ERROR]|ORDER| 27/04/2022 15:25:13 |Order manager: Unable to get active orders for Bybit and asset type usdtmarginedfutures: Param validation for 'symbol' failed on the 'linear_symbol' tag
[ERROR]|ORDER| 27/04/2022 15:25:13 |Order manager: Unable to get active orders for Bybit and asset type futures: params error!
```

Reply:
The issue here is that assets like CMF, UMF, and Futures don't support requests where the symbol is empty. 
I think this commit should fix it. [Commit](https://github.com/thrasher-corp/gocryptotrader/pull/887/commits/98ec3e4d6a805d99433f87fcfe3f7880aed68c25)

Reply:
```cmd
[ERROR]|ORDER| 02/05/2022 12:48:58 |Order manager: Unable to get active orders for Bybit and asset type usdtmarginedfutures: Param validation for 'symbol' failed on the 'linear_symbol' tag
```

Reply:
I would like to note that I also receive this:
```
[DEBUG] | REQUESTER |  05/05/2022 10:22:15  | Bybit attempt 1 request path: https://api.bybit.com/futures/private/order/list?api_key=thisIsNotARealKey&recvWindow=5000&symbol=BTCUSD&timestamp=1651710135225&sign=thisIsNotARealSignature
```
```
[DEBUG] | REQUESTER |  05/05/2022 10:19:38  | Bybit raw response: {"ret_code":10001,"ret_msg":"params error!","ext_code":"","ext_info":"","result":null,"time_now":"1651709978.623606"}
[ERROR] | ORDER |  05/05/2022 10:19:38  | Order manager: Unable to get active orders for Bybit and asset type futures: params error!
```
This is for BTC-USD asset type futures

Please add test cases for `TestGetActiveOrders` for both `Futures` and `CoinMarginedFutures`

Reply:
I figured out the root cause for an issue with the `Futures` asset. Futures symbol params should be like `BTCUSDH22`. I have added a test case for the same as well.

Reply:
@gloriousCode

Reply:
I understand your predicament with this. The API docs suggest to use the shared endpoint, which does not differentiate between the different types supported by each asset type.
They have a private API that does make the distinction: https://api2.bybit.com/v3/private/instrument/dynamic-symbol?filter=all
but I'm not going to ask you to implement a private API.

I think we will have to leave this for now as the tests are successful with the correct currencies

Reply:
@Jdpurohit Please update code and test from the suggestion below. This should fix all authentication issues and give a truer picture of available currencies per asset type via filtering. (Not perfect)

```go
// FetchTradablePairs returns a list of the exchanges tradable pairs
func (by *Bybit) FetchTradablePairs(ctx context.Context, a asset.Item) ([]string, error) {
	if !by.SupportsAsset(a) {
		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, by.Name)
	}

	switch a {
	case asset.Spot:
		allPairs, err := by.GetAllSpotPairs(ctx)
		if err != nil {
			return nil, err
		}
		pairs := make([]string, len(allPairs))
		for x := range allPairs {
			pairs[x] = allPairs[x].BaseCurrency + currency.DashDelimiter + allPairs[x].QuoteCurrency
		}
		return pairs, nil
	case asset.CoinMarginedFutures:
		allPairs, err := by.GetSymbolsInfo(ctx)
		if err != nil {
			return nil, err
		}
		m := make(map[string]struct{})
		for x := range allPairs {
			if allPairs[x].Status != "Trading" || allPairs[x].QuoteCurrency != "USD" {
				continue
			}
			symbol := allPairs[x].BaseCurrency + currency.DashDelimiter + allPairs[x].QuoteCurrency
			m[symbol] = struct{}{}
		}

		pairs := make([]string, len(m))
		target := 0
		for symbol := range m {
			pairs[target] = symbol
			target++
		}
		return pairs, nil
	case asset.USDTMarginedFutures:
		allPairs, err := by.GetSymbolsInfo(ctx)
		if err != nil {
			return nil, err
		}
		m := make(map[string]struct{})
		for x := range allPairs {
			if allPairs[x].Status != "Trading" || allPairs[x].QuoteCurrency != "USDT" {
				continue
			}

			symbol := allPairs[x].BaseCurrency +
				currency.DashDelimiter +
				allPairs[x].QuoteCurrency
			m[symbol] = struct{}{}
		}
		pairs := make([]string, len(m))
		target := 0
		for symbol := range m {
			pairs[target] = symbol
			target++
		}
		return pairs, nil
	case asset.Futures:
		allPairs, err := by.GetSymbolsInfo(ctx)
		if err != nil {
			return nil, err
		}
		pairs := make([]string, 0, len(allPairs))
		for x := range allPairs {
			if allPairs[x].Status != "Trading" {
				continue
			}

			symbol := allPairs[x].BaseCurrency + allPairs[x].QuoteCurrency
			filter := strings.Split(allPairs[x].Name, symbol)

			if len(filter) != 2 || len(filter[1]) == 0 {
				continue
			}
			pairs = append(pairs, symbol+currency.DashDelimiter+filter[1])
		}
		return pairs, nil
	}
	return nil, nil
}
```

Reply:
Matches front end: https://www.bybit.com/data/markets/contract

Reply:
<img width="700" alt="image" src="https://user-images.githubusercontent.com/9391715/170166455-f71198c8-a97c-48c9-8dea-6cb42f35dbac.png">
Also limits config.

Let me know your thoughts.

Reply:
@shazbert 
This looks like a good idea. Let me do some more testing and get back to you.
Thanks